src/lib/core/data-table/data-table.ts

```diff
@@ -92,9 +98,13 @@ export class CdkTable implements CollectionViewer {
   /** Set of templates that used as the data row containers. */
   @ContentChildren(CdkRowDef) _rowDefinitions: QueryList<CdkRowDef>;
 
-  constructor(private _changeDetectorRef: ChangeDetectorRef) {
+  constructor(private _differs: IterableDiffers,
+              private _changeDetectorRef: ChangeDetectorRef) {
     console.warn('The data table is still in active development ' +
         'and should be considered unstable.');
+
+    // TODO(andrewseguin): Add trackby function input.
+    this._dataDiffer = this._differs.find([]).create();
```

Comment:
Added some comment mentioning that we need to find an iterable differ used for array-based iterables. It's the only public API that lets us grab it, no luck instantiating one myself.